common/djangoapps/student/tests/test_models.py

```diff
@@ -403,3 +403,112 @@ def test_retire_recovery_email(self):
 
         # Assert that there is no longer an AccountRecovery record for this user
         assert len(AccountRecovery.objects.filter(user_id=user.id)) == 0
+
+@ddt.ddt
+class TestUserPostSaveCallback(SharedModuleStoreTestCase):
+    def setUp(self):
+        self.course = CourseFactory.create()
+
+    @ddt.data(*(set(CourseMode.ALL_MODES) - set(CourseMode.AUDIT_MODES)))
```

Comment:
Open to suggestions on consolidating these three very similar looking tests...

Reply:
Could you have a helper function for setting up the student with ddt.data for the different attributes each one requires and trigger the post_save callback and get course enrollment in that function?

Reply:
Do you mean another function on top of `_set_up_invited_student`? The assertions are different for all the cases so not sure that's worth consolidating

Reply:
I meant L426-434 that are somewhat repeats on the other tests, but for tests imo readability is more important than making it DRY. But yes, they are slightly different.